sdk/storage/Azure.Storage.DataMovement/src/JobPartInternal.cs

```diff
@@ -259,81 +259,60 @@ public async Task QueueChunkToChannelAsync(Func<Task> chunkTask)
         {
             // Set the status to Pause/CancellationInProgress
-            if (DataTransferStatus.PauseInProgress == _dataTransfer.TransferStatus)
+            if (DataTransferState.Pausing == _dataTransfer.TransferStatus.State)
             {
                 // It's possible that the status hasn't propagated down to the job part
                 // status yet here since we pause from the data transfer object.
-                await OnTransferStatusChanged(DataTransferStatus.PauseInProgress).ConfigureAwait(false);
+                await OnTransferStateChangedAsync(DataTransferState.Pausing).ConfigureAwait(false);
             }
             else
             {
                 // It's a cancellation if a pause wasn't called.
-                await OnTransferStatusChanged(DataTransferStatus.CancellationInProgress).ConfigureAwait(false);
+                await OnTransferStateChangedAsync(DataTransferState.Stopping).ConfigureAwait(false);
             }
             await CleanupAbortedJobPartAsync().ConfigureAwait(false);
         }
 
         /// <summary>
         /// To change all transfer statues at the same time
         /// </summary>
-        /// <param name="transferStatus"></param>
-        internal async Task OnTransferStatusChanged(DataTransferStatus transferStatus)
+        /// <param name="transferState"></param>
+        internal async Task OnTransferStateChangedAsync(DataTransferState transferState)
         {
-            bool statusChanged = false;
-            lock (_statusLock)
-            {
-                if (transferStatus != DataTransferStatus.None
-                    && JobPartStatus != transferStatus)
-                {
-                    statusChanged = true;
-                    JobPartStatus = transferStatus;
-                }
-            }
-            if (statusChanged)
+            if (JobPartStatus.TrySetTransferStateChange(transferState))
             {
                 // Progress tracking, do before invoking the event below
-                if (transferStatus == DataTransferStatus.InProgress)
+                if (transferState == DataTransferState.InProgress)
                 {
                     _progressTracker.IncrementInProgressFiles();
                 }
-                else if (transferStatus == DataTransferStatus.Completed)
-                {
-                    _progressTracker.IncrementCompletedFiles();
-                }
-                else if (transferStatus == DataTransferStatus.CompletedWithSkippedTransfers)
-                {
-                    _progressTracker.IncrementSkippedFiles();
-                }
-                else if (transferStatus == DataTransferStatus.CompletedWithFailedTransfers)
-                {
-                    _progressTracker.IncrementFailedFiles();
-                }
 
-                if (JobPartStatus == DataTransferStatus.Completed)
+                if (JobPartStatus.HasCompletedSuccessfully)
```

Comment:
Style but should this be `else if` given it's mutually exclusive with the previous condition?

Reply:
done